### ISSUE TITLE
[Bug]: Plugin Mojo crashes in Quarkus Dev Mode

### ISSUE THREAD

**Issue by ivu-mawi** [ascopes/protobuf-maven-plugin#488]
### Brief description

My projects use the Quarkus framework. When debugging, it is recommended to use Quarkus dev mode (`mvn quarkus:dev`)

Currently, this plugin crashes in that dev mode with the attached error.

Also happens with older versions of either this plugin or Quarkus, so I'm not sure why this worked earlier or when it stopped exactly. Maybe `resolve(null)` did not always throw?

I think I've tracked it down to `mojoExecution.executionId` being `null` during Quarkus Dev mode (see Screenshot below). Would you say this is more of a Quarkus bug?

**Edit:** Works with v2.6.2 which is before #421

### Steps to reproduce

1. Download the Quarkus example https://github.com/quarkusio/quarkus-quickstarts/tree/main/getting-started
2. Add the Plugin to build plugin executions in pom.xml
3. Add any example .proto file (into the default proto source folder)
4. Run `./mvnw -e quarkus:dev`

### Error message

[error.txt](https://github.com/user-attachments/files/18094479/error.txt)

### Operating system

Windows

### JDK version

Temurin 21.0.5

### Maven version

3.9.9

### Plugin version

2.7.2

### Protoc version

unknown

### Additional details

![Screenshot 2024-12-11 122325](https://github.com/user-attachments/assets/5bfc66b4-2756-413e-9bab-334bfcf31652)

### Would you like to contribute a bugfix?

- [ ] Yes, I'd like to contribute a bugfix!

**Comment by ascopes** [comment]
Hmm... that is interesting!

Do we know if this is expected behaviour from Quarkus devs for the execution ID to be null?

**Comment by ascopes** [comment]
I've raised a PR. Please can you build it on your side and verify it fixes the issue for you?

**Comment by ivu-mawi** [comment]
Thanks for the quick fix!

I've tried logging the `goal` and `executionId` during quarkus dev mode and it turns out that `goal` is in fact always populated, just `executionId` is always null. I'm not sure why my debugger showed `goal = null`.

So technically not a Quarkus bug then, I'm thinking. Thanks again!

PS: Took me a second to figure out how to build the plugin myself without the integration tests, because they took long and then failed due to a C compiler warning I believe. I ended up doing `-Dmaven.test.skip=true -Dinvoker.skip=true`
